### PR TITLE
Fix & Optimize CI config

### PR DESCRIPTION
**Pull request by 6543**

### CODE REVIEW

File: .woodpecker.yml [hunk 1]
```diff
@@ -216,11 +218,11 @@ pipeline:
 
   # using https://github.com/pksunkara/cargo-workspaces
   publish_to_crates_io:
-    image: clux/muslrust:1.67.0
+    image: *muslrust_image
     commands:
       - cargo install cargo-workspaces
       - cp -r migrations crates/db_schema/
-      - cargo login "$CARGO_TOKEN"
+      - cargo login "$CARGO_API_TOKEN"
```
Comment (6543) [inline .woodpecker.yml:225]: make sure the secret is called "cargo_api_token" or "CARGO_API_TOKEN" in woodpecker too

File: .woodpecker.yml [hunk 2]
```diff
@@ -1,18 +1,20 @@
 # TODO: The when: platform conditionals aren't working currently
 # See https://github.com/woodpecker-ci/woodpecker/issues/1677
 
+variables:
+  - &muslrust_image 'clux/muslrust:1.67.0'
+
+clone:
+  git:
+    image: woodpeckerci/plugin-git
+    settings:
```
Comment (6543) [inline .woodpecker.yml:10]: https://woodpecker-ci.org/docs/next/usage/pipeline-syntax#git-submodules

File: .woodpecker.yml [hunk 3]
```diff
@@ -1,18 +1,20 @@
 # TODO: The when: platform conditionals aren't working currently
 # See https://github.com/woodpecker-ci/woodpecker/issues/1677
 
+variables:
```
Comment (6543) [inline .woodpecker.yml:4]: https://woodpecker-ci.org/docs/next/usage/advanced-yaml-syntax

**Comment by 6543** [comment]
PS: I'll have a look into multiarch build for your docker ... so it does not matter at witch arch the agent is running in order to generate arm/x86 images

File: .woodpecker.yml (hunk 3 above)
Comment (dessalines) [reply .woodpecker.yml:4]: Thanks! This is def helpful.

File: .woodpecker.yml (hunk 1 above)
Comment (dessalines) [reply .woodpecker.yml:225]: Checked, that's good.

**Review by dessalines — APPROVED** [review]
Thanks for these! I'll apply that variable technique to the other woodpecker projects we have set up.

**Comment by 6543** [comment]
ok got cross compile working ->  #2804

so now it doesn't matter if you run the docker-buildx step on x86 or arm64 :)

I'll keep mentioned pull as draft until this here got merged

**Comment by 6543** [comment]
tldr: one step to compile for both arch

**Comment by 6543** [comment]
@dessalines CI passed

**Review by Nutomic — APPROVED** [review]
